app/components/SimulationCanvas.tsx

```diff
@@ -9,40 +9,94 @@ import LightsSetup from './LightSetup';
 import ShipController from './ShipController';
 import CameraController from './CameraController';
 import ManualControls from './ManualControls';
+import SimulationControls from './SimulationControls';
+import { SimulationControlsParams } from './SimulationControls';
+
+//CONSTANTS
+// This is the actual time the simulation will take reagardless of the travel time
+const SIM_DURATION_SECONDS = 30;
 
 // SimulationCanvas component where everything is wrapped inside the Canvas
 export default function SimulationCanvas() {
   const shipRef = useRef<THREE.Mesh>(null!); // Shared ref so that we have the latest ref for both the Ship and Camera.
+  const animationFrameRef = useRef<number | null>(null); // For canceling animation
+
   const [isManualControl, setIsManualControl] = useState(false);
+  const [simulatedYear, setSimulatedYear] = useState(0);
+  const [isJourneyActive, setIsJourneyActive] = useState(false);
+  const [elapsedTime, setElapsedTime] = useState(0);
+  const [timeCompression, setTimeCompression] = useState(0);
+
+  const baseYear = new Date().getFullYear();
+
+
 
   const toggleControlMode = () => {
     setIsManualControl(prev => !prev);
   };
 
+  const handleJourneyStart = ({ destination, speed, travelTime }: SimulationControlsParams) => {
+    console.log('Starting journey to', destination, 'at', speed, '% of light');
+    console.log('It\'s going to take', travelTime, 'years to reach our destination');
+
+    // Setting Camera Control to Auto
+    if (isManualControl) setIsManualControl(false);
+
+    // Simulation logic
+    const startTime = performance.now();
+    const endTime = startTime + SIM_DURATION_SECONDS * 1000;
+    // Calulating for X here (1s = Xyrs), X(years/ms) = travelTime/simDurationInMs 
+    const yearsPerMs = travelTime / (SIM_DURATION_SECONDS * 1000);
+
+    const compressionFactor = Math.floor(yearsPerMs * 1000); // years per second
+    setTimeCompression(compressionFactor);
+    console.log(`travelTime: ${travelTime}, yearsPerMs: ${yearsPerMs}, compressionFactor: ${compressionFactor}`);
+
+  
+    setIsJourneyActive(true);
+    setElapsedTime(0); // Reset Elapsed/Animation time
+    setSimulatedYear(0); // Reset year at start
+  
+    const animate = (now: number) => {
+      if (now >= endTime) {
+        setSimulatedYear(travelTime);
+        setElapsedTime(SIM_DURATION_SECONDS);
+        setIsJourneyActive(false);
+        animationFrameRef.current = null;
+        return;
+      }
+  
+      const elapsed = now - startTime;
+      const currentYear = parseFloat((elapsed * yearsPerMs).toFixed(2));
+      setSimulatedYear(currentYear);
+      setElapsedTime(Math.floor(elapsed / 1000));
+  
+      animationFrameRef.current = requestAnimationFrame(animate);
+    };
+  
+    animationFrameRef.current = requestAnimationFrame(animate);
+  };
+
```

Comment:
We should have a cleanup function to clear the animation if the component is unmounted(it could lead to have lingering animations), to avoid any performance issues or errors in the future.